src/visualizer/vertex/Vertex.hpp

```diff
@@ -3,22 +3,35 @@
 //
 
 #pragma once
-
+#include <QMouseEvent>
 #include <QPainter>
 #include <QWidget>
+#include <cmath>
+#include <intern/QtDefines.hpp>
 
 namespace age::visualizer {
 class Vertex : public QWidget {
   Q_OBJECT
 public:
   explicit Vertex(int x, int y, QWidget* pParent) noexcept;
+  explicit Vertex(QPoint const& point, QWidget* pParent) noexcept;
+signals:
```

Comment:
NL between access specifiers. Even though `signals` is just `public`

```suggestion
  explicit Vertex(QPoint const& point, QWidget* pParent) noexcept;
  
signals:
```